lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -528,14 +582,80 @@ export default class BitcoinProcessor {
       }
     }
 
+    if (this.isBatchBoundary(block)) {
+
+      // Compute the transaction fees for sampled transactions of this batch
+      const sampledTransactionIds = this.transactionSampler.getSample();
+      const sampledTransactionFees = new Array();
+      for (let transactionId of sampledTransactionIds) {
+        const transactionFee = await this.getTransactionFeeInSatoshi(transactionId);
+        sampledTransactionFees.push(transactionFee);
+      }
+
+      const batchId = this.getBatchId(block);
+      await this.quantileCalculator.add(batchId, sampledTransactionFees);
+
+      // Reset the sampler for the next batch
+      this.transactionSampler.clear();
+    }
+
     return blockHash;
   }
 
+  private isBatchBoundary (block: number): boolean {
+    return (block + 1) % this.proofOfFeeConfig.transactionFeeQuantileConfig.batchSizeInBlocks === 0;
+  }
+
+  private getBatchId (block: number): number {
```

Comment:
Design sanity: is this approach resilient to rollbacks? I have a suspicion that rollback needs to be handled. 2 strategies: either rollback when reorg happens; or only store samples that are older "enough" that reorg is mathematically impossible to occur.

Reply:
Good catch. I missed rollbacks. I went with the first approach since the second approach relies on probabilistic assumptions regarding rollbacks. The current code is designed to work for any rollback value.

Reply:
To also simplify the code, I rollback to a "batch" boundary when I can reset state cleanly. The assumption here is that rollbacks are rare events. If we know otherwise, we can avoid rollbacks to batch boundaries for "recent" rollbacks - I would suggest implementing this only if we have a data point suggesting the current solution is expensive.